src/Nethermind/Nethermind.BeamWallet/Modules/Transfer/TransferModule.cs

```diff
@@ -77,14 +86,104 @@ private async Task UpdateBalanceAsync()
+            _balanceEth = balance.Value;
             _transferWindow.Remove(_balanceLabel);
             _balanceLabel = new Label(65, 1, $"Balance: {_balanceEth} ETH");
 
             _transferWindow.Add(_balanceLabel);
         }
+        
+        private async Task<decimal?> GetBalanceAsync()
+        {
+            var result = await _ethJsonRpcClientProxy.eth_getBalance(_address);
+            if (!result.IsValid || !result.Result.HasValue)
+            {
+                return null;
+            }
+            return WeiToEth(decimal.Parse(result.Result.ToString()));
+        }
+        
+        private async Task GetAverageGasPriceAsync()
+        {
+            _blockNumberLabel = new Label(1, 11, "eth_getBlockByNumber: calling...");
+            _transferWindow.Add(_blockNumberLabel);
+            RpcResult<BlockModel<TransactionModel>> result;
+            do
+            {
+                result = await _ethJsonRpcClientProxy.eth_getBlockByNumberWithTransactionDetails(
+                    BlockParameterModel.Latest,
+                    true);
+                await Task.Delay(2000);
+            } while (!result.IsValid);
+            
+            _latestBlock = result.Result;
+            UInt256 sum = 0;
+            if (_latestBlock.Transactions.Count == 0)
+            {
+                _averageGasPrice = _gasPrice;
+                _transferWindow.Remove(_blockNumberLabel);
+                _blockNumberLabel = new Label(1, 11, $"eth_getBlockByNumber: {_latestBlock.Number}, average gas price: " +
+                                                     $"{_averageGasPrice} WEI");
+                _transferWindow.Add(_blockNumberLabel);
+            }
+            
+            int transactionCount = 0;
+            foreach (var transaction in _latestBlock.Transactions)
+            {
+                if (transaction.GasPrice > 100)
+                {
+                    sum += transaction.GasPrice;
+                    transactionCount++;
+                }
+            }
+
+            _averageGasPriceNumber = transactionCount > 0 ? (long)sum / transactionCount : (long)sum;
+            _transferWindow.Remove(_blockNumberLabel);
+            _blockNumberLabel = new Label(1, 11, $"eth_getBlockByNumber: {_latestBlock.Number}, average gas price: " +
+                                                $"{_averageGasPriceNumber} WEI");
+            _transferWindow.Add(_blockNumberLabel);
+            _averageGasPrice = UInt256.Parse(_averageGasPriceNumber.ToString());
+        }
+
+        private async Task GetTransactionCountAsync()
+        {
+            _nonceLabel = new Label(1, 13, "eth_getTransactionCount: calling...");
+            _transferWindow.Add(_nonceLabel);
+            RpcResult<UInt256?> result;
+            do
+            {
+                result = await _ethJsonRpcClientProxy.eth_getTransactionCount(_address);
+                await Task.Delay(3000);
```

Comment:
waiting 3 seconds even when it works immediately, also waiting additional 3 seconds after any success